pkg/daemon/reader.go

```diff
@@ -206,7 +211,9 @@ func readerPollLoop(
 	pbp := PeriodBetweenPolls
 	var lastError time.Time
 	var candidateForRemove bool
-	var currentlyLoadedCard state.Token
+	// keep track of core switch for menu reset
+	var lastCoreName string = mrextConfig.MenuCore
```

Comment:
we need to keep track when we switch from a game to menu to reset our internal state. The only way i found that was suitable was to track was to create a local variable that will allow me to be either the menu core or something else. Updating the variable each loop allows me to run the clean condition only once.
If we had a sort of event of when the core reset that would better.